### PR TITLE
Use cmake to build openni2 on windows

### PR DESCRIPTION
**Pull request by vinhloi**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,8 +2,12 @@ cmake_minimum_required (VERSION 2.8)
...
-add_subdirectory(libusbx)
+if(APPLE)
+    add_subdirectory(libusbx)
+endif(APPLE)
+
 add_subdirectory(ThirdParty/LibJPEG)
 add_subdirectory(ThirdParty/PSCommon/XnLib)
 add_subdirectory(Source)
 add_subdirectory(Source/Drivers/PS1080)
+add_subdirectory(Source/Drivers/Kinect)
```
Comment (willkelleher) [inline CMakeLists.txt]: We'll need an if(WIN32) around this to avoid breaking the Mac builds, I think.

File: Source/Core/OniContext.cpp [hunk 2]
```diff
@@ -118,8 +118,8 @@ OniStatus Context::initialize()
 	}
 
 	xnLogVerbose(XN_LOG_MASK_ALL, "Using '%s' as driver path", ONI_DEFAULT_DRIVERS_REPOSITORY);
-	// Use default path
-	rc = loadLibraries(ONI_DEFAULT_DRIVERS_REPOSITORY);
```
Comment (willkelleher) [inline Source/Core/OniContext.cpp:122]: This seems wrong. Can we change the driver path with the .ini file instead?

File: ThirdParty/PSCommon/XnLib/CMakeLists.txt [hunk 3]
```diff
@@ -91,3 +92,4 @@ set (XNLIB_SRC_OS
     )
 endif()
 add_library(XnLib STATIC ${XNLIB_CORE_SRC} ${XNLIB_LOG_SRC} ${XNLIB_COMMON_OS_SRC} ${XNLIB_SRC_OS})
+target_link_libraries(XnLib ws2_32.lib Setupapi.lib)
```
Comment (willkelleher) [inline ThirdParty/PSCommon/XnLib/CMakeLists.txt]: This needs to be within if(WIN32)

**Comment by vinhloi** [comment]
@willkelleher : I have updated the cmake files. A small notice is that the Kinect driver in our fork doesn't support ImageRegistrationMode. Should we sync our fork with the original OpenNI2?

**Comment by dscherba** [comment]
@vinhloi can you elaborate more on the kinect (k4w or 'legacy' xbox 360 kinect or both) ImageRegistrationMode thing?

Is this available upstream (or was it renamed)?

Does it default to on (if so this might be fine)?

Basically, we need this functionality enabled in some manner, otherwise we can't extract as the depth and color won't be properly aligned.

**Comment by willkelleher** [comment]
I thought we had this?

**Comment by vinhloi** [comment]
@dscherba : it is available upstream. To be more precise, it is on commit f77cd9d4f8bd1cf49d7eb1d3715782441bd835ed , 5/8/2013 on the master branch of the OpenNI2 repo.
And it is default to off, which results in the depth and color not aligned when I run the PersonifyTest app with Kinect camera.

**Comment by dscherba** [comment]
@vinhloi thanks for more detail... this is K4W stuff (which is why @willkelleher didn't see/need it -- this camera isn't supported on Mac!). Yes, we should rebase any changes/build stuff to use an ONI2 version that supports what we need (if we can get everything we want w/o any Personify customizations, e.g., for image registration, for motor adjustments, etc...) that's the ideal.

**Comment by svensht2** [comment]
bringing this back to life @dscherba @willkelleher

File: CMakeLists.txt [hunk 4]
```diff
@@ -2,7 +2,13 @@ cmake_minimum_required (VERSION 2.8)
 
 project(OpenNI2)
 
-add_subdirectory(libusbx)
+add_definitions(-UUNICODE)
+add_definitions(-U_UNICODE)
+
+if(APPLE)
```
Comment (willkelleher) [inline CMakeLists.txt:8]: might be more accurate to do if(NOT WIN32), because I think Linux will use libusbx, too.

**Comment by willkelleher** [comment]
Tested on Mac, seems to work fine.  We can deal with Linux support in a future PR, I guess.

**Comment by willkelleher** [comment]
:+1:

**Comment by svensht2** [comment]
yes, lets have noah do it, it should fairly straightforward

**Pull request merged**